cli/src/main/java/com/box/l10n/mojito/cli/command/utils/SarifFileGenerator.java

```diff
@@ -0,0 +1,145 @@
+package com.box.l10n.mojito.cli.command.utils;
+
+import com.box.l10n.mojito.cli.GitInfo;
+import com.box.l10n.mojito.cli.command.checks.CheckerRuleId;
+import com.box.l10n.mojito.cli.command.checks.CliCheckResult;
+import com.box.l10n.mojito.cli.command.extraction.AssetExtractionDiff;
+import com.box.l10n.mojito.okapi.extractor.AssetExtractorTextUnit;
+import com.box.l10n.mojito.sarif.builder.SarifBuilder;
+import com.box.l10n.mojito.sarif.model.Location;
+import com.box.l10n.mojito.sarif.model.ResultLevel;
+import com.box.l10n.mojito.sarif.model.Sarif;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+
+@Component
+public class SarifFileGenerator {
+
+  static Logger logger = LoggerFactory.getLogger(SarifFileGenerator.class);
+
+  private final GitInfo gitInfo;
+
+  private final String infoUri;
+
+  private final String[] supportedFileExtensions;
+
+  @Autowired
+  SarifFileGenerator(
+      @Value("${l10n.extraction-check.sarif.infoUri:}") String infoUri,
+      @Value(
+              "#{'${l10n.extraction-check.sarif.commentsChecks.lineUpdate.supportedFileExtensions:py,xml}'.split(',')}")
```

Comment:
Shouldn't `ts`, `js` be included in defaults here

Reply:
Ah, I misunderstood the reasoning here with `supportedFileExtensions`, this is file extensions where the comment is above the call to the the i18n function? Maybe rename to `lineUpdateFileExtensions` if this is the case?